homeassistant/components/ecobee/notify.py

```diff
@@ -1,31 +1,35 @@
 """Support for Ecobee Send Message service."""
-import voluptuous as vol
 
-from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.notify import ATTR_TARGET, BaseNotificationService
 
-from .const import CONF_INDEX, DOMAIN
+from homeassistant.exceptions import RequiredParameterMissing
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_INDEX, default=0): cv.positive_int}
-)
+from .const import DOMAIN
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the Ecobee notification service."""
+    if discovery_info is None:
+        return None
+
     data = hass.data[DOMAIN]
-    index = config.get(CONF_INDEX)
-    return EcobeeNotificationService(data, index)
+    return EcobeeNotificationService(data.ecobee)
 
 
 class EcobeeNotificationService(BaseNotificationService):
     """Implement the notification service for the Ecobee thermostat."""
 
-    def __init__(self, data, thermostat_index):
+    def __init__(self, ecobee):
         """Initialize the service."""
-        self.data = data
-        self.thermostat_index = thermostat_index
+        self.ecobee = ecobee
 
     def send_message(self, message="", **kwargs):
         """Send a message."""
-        self.data.ecobee.send_message(self.thermostat_index, message)
+        targets = kwargs.get(ATTR_TARGET, None)
+
+        if not targets:
+            raise RequiredParameterMissing(["target"])
```

Comment:
We normally raise `ValueError` with a message as argument on bad user input in service handlers.

Reply:
Fixed in new commit